clientd3d/d3drender_skybox.c

```diff
@@ -137,14 +126,16 @@ static unsigned char gSkyboxBGRA[] =
 // Interfaces
 
 static void D3DRenderBackgroundsLoad(char* pFilename, int index);
-static void D3DRenderBackgroundSet(ID background);
+static bool D3DRenderBackgroundSet(ID background);
 
 // Implementations
 
 /**
-* Initialize the skybox rendering process.
+* Initialize the skybox with the current room.
+* Return true the background has changed and false otherwise.
+* If the background has changed, the caller should trigger a redraw all.
 */
-void D3DRenderSkyBoxBegin()
+bool D3DRenderSkyBoxBegin(room_type* current_room)
```

Comment:
current_room is only used to get bkgnd, so how about just passing in bkgnd instead?